### PR TITLE
Revert "lock db at high level **per image** (#3201)"

### PR DESCRIPTION
**Pull request by TurboGit**
This reverts commit 979787652e0106d15023c879271f498a20b2fa87.

### CODE REVIEW

**Comment by TurboGit** [comment]
Please DO NOT MERGE. This is for testing.

File: src/common/film.c [hunk 1]
```diff
@@ -200,6 +200,7 @@ int dt_film_new(dt_film_t *film, const char *directory)
                                 -1, &stmt, NULL);
     DT_DEBUG_SQLITE3_BIND_TEXT(stmt, 1, datetime, -1, SQLITE_STATIC);
     DT_DEBUG_SQLITE3_BIND_TEXT(stmt, 2, directory, -1, SQLITE_STATIC);
+    dt_pthread_mutex_lock(&darktable.db_insert);
```
Comment (TurboGit) [inline src/common/film.c]: I'm wondering if the issue is now there. A lock has been removed.

Comment (parafin) [reply src/common/film.c]: I'm not sure how this code can race, but sure, it's worth a try to bring back this lock. Though there are a lot of other places where DB is modified, but no lock is present.

**Comment by TurboGit** [comment]
Closing, the fix was merged in commit 344ce8c34d3.

File: src/common/film.c (hunk 1 above)
Comment (TurboGit) [reply src/common/film.c]: I didn't re-introdouce this lock as after analysis and testing it is not necessary to me.

**Comment by jenshannoschwalm** [comment]
There is _dev_merge_history that looks very suspicious to me.

You get a list of row id's that are **not** kept inside a transcaction. So another thread might change database lines between collection the id's and processing data that might now be incorrectly referenced.

**Comment by TurboGit** [comment]
@jenshannoschwalm : I have merged the proper fix (I think), this was only for testing purpose.

**Pull request closed without merging**